test/integration/test_api_logging.py

```diff
@@ -0,0 +1,337 @@
+        assert stdout_output == ""
+        assert stderr_output != ""
+        assert create_test_regex(logging.ERROR).search(stderr_output) is not None
+        assert create_test_regex(logging.INFO).search(stderr_output) is None
+        main_log_glob = list(luigi_output.glob("**/main.log"))
+        assert main_log_glob == []
+
+
+def test_luigi_log_level_error_multiple_calls_and_basic_logging_info(
+    capfd, custom_logging, luigi_output
+):
+    """
+    This test checks if setting luigi log level to level ERROR and global log level to INFO
+    prints only error messages to stderr, but no INFO messages; even when running a task multiple times.
+    Note: This test does not work with `pytest -s ...`
+    """
+    with custom_logging(log_level=logging.INFO) as error_log:
+
+        logger_infos_before = create_logger_infos()
+        result = dummy_api_command(log_level="ERROR", use_job_specific_log_file=False)
+        result = dummy_api_command(log_level="ERROR", use_job_specific_log_file=False)
+        logger_infos_after = create_logger_infos()
+        assert_loggers_are_equal(logger_infos_after, logger_infos_before)
+        stdout_output, stderr_output = capfd.readouterr()
+        assert stdout_output == ""
+        assert stderr_output != ""
+        assert create_test_regex(logging.ERROR).search(stderr_output) is not None
+        assert create_test_regex(logging.INFO).search(stderr_output) is None
+        assert 2 == stderr_output.count("DUMMY LOGGER ERROR")
+        main_log_glob = list(luigi_output.glob("**/main.log"))
+        assert main_log_glob == []
+
+
+def test_luigi_use_job_specific_log_file_and_basic_logging_error(
+    capfd, custom_logging, luigi_output
+):
+    """
+    This test checks if setting luigi log level to None and global log level to ERROR, and using flag
+    `use_job_specific_log_file`
+    prints no error messages to stderr/stdout, but writes the log messages to `main.log`.
+    Note: This test does not work with `pytest -s ...`
+    """
+
+    with custom_logging(log_level=logging.ERROR) as error_log:
+
+        logger_infos_before = create_logger_infos()
+        result = dummy_api_command(log_level=None, use_job_specific_log_file=True)
+        logger_infos_after = create_logger_infos()
+        assert_loggers_are_equal(logger_infos_after, logger_infos_before)
+        stdout_output, stderr_output = capfd.readouterr()
+
+        # Python3.12 (and later?) prints a deprecation warning.
+        if sys.version_info[1] < 12:
+            assert stderr_output == ""
+        else:
+            stderr_output_lines = stderr_output.split("\n")
+            assert len(stderr_output_lines) == 3
+            assert (
+                " is multi-threaded, use of fork() may lead to deadlocks in the child"
+                in stderr_output_lines[0]
+            )
+            assert stderr_output_lines[1] == "  self.pid = os.fork()"
+            assert stderr_output_lines[2] == ""
```

Comment:
Seems this not happening with pytests, see https://github.com/exasol/integration-test-docker-environment/actions/runs/14365053293/job/40276571989?pr=473

Reply:
interesting, lets discuss this offline

maybe of some interest
https://github.com/pytest-dev/pytest-forked/issues/85
https://github.com/pytest-dev/pytest/pull/10894

Reply:
Ok, pytest consumes warning. Using pytests fixture `recwarn` showed the warning.

Reply:
This means we can remove the special logic for Python 3.12 and above